api/src/routes/helpers/auth-helpers.ts

```diff
@@ -11,25 +16,375 @@ export const findOrCreateUser = async (
   fastify: FastifyInstance,
   email: string
 ): Promise<{ id: string; acceptedPrivacyTerms: boolean }> => {
-  // TODO: handle the case where there are multiple users with the same email.
-  // e.g. use findMany and throw an error if more than one is found.
-  const existingUser = await fastify.prisma.user.findMany({
+  const existingUsers = await fastify.prisma.user.findMany({
     where: { email },
-    select: { id: true, acceptedPrivacyTerms: true }
-  });
-  if (existingUser.length > 1) {
-    fastify.Sentry.captureException(
-      new Error(
-        `Multiple user records found for: ${existingUser.map(user => user.id).join(', ')}`
-      )
-    );
+    select: {
+      id: true,
+      acceptedPrivacyTerms: true
+    }
+  });
+
+  if (existingUsers.length > 1) {
+    const userIds = existingUsers.map(user => user.id);
+    fastify.log.info(`Multiple user records found for: ${userIds.join(', ')}`);
+
+    try {
+      const users = await fastify.prisma.user.findMany({
+        where: { email }
+      });
+      // Merge the duplicate users
+      const mergedData = mergeDuplicateUsers(users);
+      const oldestUserId = mergedData.id;
+
+      const userWithoutId = omit(mergedData, ['id']);
+
+      // Update the oldest user with merged data
+      await fastify.prisma.user.update({
+        where: { id: oldestUserId },
+        data: removeNulls(userWithoutId) as Parameters<
+          typeof fastify.prisma.user.update
+        >[0]['data']
+      });
+
+      const duplicateUserIds = userIds.filter(i => i !== oldestUserId);
```

Comment:
```suggestion
      const duplicateUserIds = userIds.filter(userId => userId !== oldestUserId);
```
Filter item value could be rename to make it more readable